src/plugins/intel_gpu/src/runtime/debug_configuration.cpp

```diff
@@ -238,16 +246,31 @@ const debug_configuration *debug_configuration::get_instance() {
 #endif
 }
 
-bool debug_configuration::is_dumped_layer(const std::string& layerName, bool is_output) const {
+bool debug_configuration::is_dumped_layer(const std::string& layer_name, bool is_output) const {
 #ifdef GPU_DEBUG_CONFIG
     if (is_output == true && dump_layers_result == 1 &&
-        (layerName.find("constant:") == std::string::npos))
+        (layer_name.find("constant:") == std::string::npos))
         return true;
     if (dump_layers.empty() && dump_layers_result == 0)
         return true;
 
+    auto is_match = [](const std::string& layer_name, const std::string& pattern) -> bool {
+        // Check pattern from exec_graph
+        size_t pos = layer_name.find(':');
+        auto exec_graph_name = layer_name.substr(pos + 1, layer_name.size());
+        auto upper_exec_graph_name = std::string(exec_graph_name.length(), '\0');
+        std::transform(exec_graph_name.begin(), exec_graph_name.end(), upper_exec_graph_name.begin(), ::toupper);
+        auto upper_pattern = std::string(pattern.length(), '\0');
+        std::transform(pattern.begin(), pattern.end(), upper_pattern.begin(), ::toupper);
+        if (upper_exec_graph_name.compare(upper_pattern) == 0) {
+            return true;
+        }
+        // Check pattern with regular expression
+        std::regex re(pattern, std::regex::icase);
```

Comment:
1) Oh it seems that icase enables to match regardless cases
2) But seems that in some version of gcc, icase has a bug 
3) So anyway it seems that transform to upper looks better
4) But then, we don't need icase?

Reply:
Okay, I see, I updated to use upper with transformer.
BTW what means "we don't need icase"?  Our expectation is insensitive case and std::regex with std::regex::icase define it runwithout regard to caseignore.